lib/pbench/test/unit/agent/test_tool_meister.py

```diff
@@ -0,0 +1,202 @@
+            c = subprocess.CompletedProcess(args, returncode=0, stdout=b"", stderr=None)
+            assert all(
+                x in c.args[0]
+                for x in [
+                    "tar_path",
+                    "--create",
+                    "--xz",
+                    "--force-local",
+                    f"--file={tar_file}",
+                ]
+            )
+            return c
+
+        monkeypatch.setattr(subprocess, "run", fake_run)
+
+        cp = tool_meister._create_tar(tmp_dir, pathlib.Path(tar_file))
+        assert cp.returncode == 0
+        assert cp.stdout == b""
+
+    @staticmethod
+    def test_create_tar_ignore_warnings(tool_meister, monkeypatch):
+        """Test creating tar with warning=none option specified"""
+
+        def fake_run(*args, **kwargs):
+            if "--warning=none" in args[0]:
+                return subprocess.CompletedProcess(
+                    args,
+                    returncode=0,
+                    stdout=b"No error after --warning=none",
+                    stderr=None,
+                )
+            else:
+                return subprocess.CompletedProcess(
+                    args, returncode=1, stdout=b"Some error running tar", stderr=None
+                )
+
+        monkeypatch.setattr(subprocess, "run", fake_run)
+
+        cp = tool_meister._create_tar(tmp_dir, pathlib.Path(tar_file))
+        assert cp.returncode == 0
+        assert cp.stdout == b"No error after --warning=none"
+
+    @staticmethod
+    def test_create_tar_failure(tool_meister, monkeypatch, caplog):
+        """Test tar creation failure"""
+
+        def fake_run(*args, **kwargs):
+            return subprocess.CompletedProcess(
+                args,
+                returncode=1,
+                stdout=b"Some error running tar command, empty tar creation failed",
+                stderr=None,
+            )
+
+        monkeypatch.setattr(subprocess, "run", fake_run)
+
+        cp = tool_meister._create_tar(tmp_dir, pathlib.Path(tar_file))
+        assert cp.returncode == 1
+        assert cp.stdout == b"Some error running tar command, empty tar creation failed"
+        assert (
+            "Tarball creation failed with 1 (stdout 'Some error running tar"
+            f" command, empty tar creation failed') on {tmp_dir}: Re-trying now"
+            in caplog.text
+        )
+
+
+class TestSendDirectory:
+    """Test ToolMeister._send_directory()'s use of ._create_tar()"""
+
+    # Record all the mock functions called by a test
+    functions_called = []
+
+    directory = tmp_dir / f"{tm_params['hostname']}"
+
+    @staticmethod
+    def fake_create_tar(returncode: int, stdout: bytes):
+        def f(directory: pathlib.Path, tar_file: pathlib.Path):
+            return subprocess.CompletedProcess(
+                args=[], returncode=returncode, stdout=stdout, stderr=None
+            )
+
+        __class__.functions_called.append("fake_create_tar")
```

Comment:
Uh...eh-hem...shouldn't this call to `append()` be inside the mock function (i.e., before line 134)??  😆